src/json_api.rs

```diff
@@ -1,115 +0,0 @@
-///
-/// Any thumbnail or c2pa data will be written to data_dir if provided
-pub fn read_ingredient_file(path: &str, data_dir: &str) -> Result<String> {
-    Ok(Ingredient::from_file_with_folder(path, data_dir)
-        .map_err(Error::from_c2pa_error)?
-        .to_string())
-}
-
-/// Adds a manifest to the source file and writes the result to the destination file.
-/// Also returns the binary manifest data for optional cloud storage
-/// A manifest definition must be supplied
-/// Signer information must also be supplied
-///
-/// Any file paths in the manifest will be read relative to the source file
-pub fn sign_file(
-    source: &str,
-    dest: &str,
-    manifest_json: &str,
-    signer_info: &SignerInfo,
-    data_dir: Option<String>,
-) -> Result<Vec<u8>> {
-    let mut manifest = Manifest::from_json(manifest_json).map_err(Error::from_c2pa_error)?;
-
-    // if data_dir is provided, set the base path for the manifest
-    if let Some(path) = data_dir {
-        manifest
-            .with_base_path(path)
-            .map_err(Error::from_c2pa_error)?;
-    }
-
-    // If the source file has a manifest store, and no parent is specified, treat the source's manifest store as the parent.
-    if manifest.parent().is_none() {
-        let source_ingredient = Ingredient::from_file(source).map_err(Error::from_c2pa_error)?;
-        if source_ingredient.manifest_data().is_some() {
-            manifest
-                .set_parent(source_ingredient)
-                .map_err(Error::from_c2pa_error)?;
-        }
-    }
-
-    let signer = signer_info.signer()?;
-    manifest
-        .embed(&source, &dest, &*signer)
-        .map_err(Error::from_c2pa_error)
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use std::{fs::remove_dir_all, path::PathBuf};
-
-    /// returns a path to a file in the fixtures folder
-    pub fn test_path(path: &str) -> String {
-        let base = env!("CARGO_MANIFEST_DIR");
-        format!("{}/{}", base, path)
-    }
-
-    #[test]
-    fn test_verify_from_file_no_base() {
-        let path = test_path("tests/fixtures/C.jpg");
-        let result = read_file(&path, None);
-        assert!(result.is_ok());
-        let json_report = result.unwrap();
-        println!("{}", json_report);
-        assert!(json_report.contains("C.jpg"));
-        assert!(!json_report.contains("validation_status"));
-    }
-
-    #[test]
-    fn test_read_from_file_with_base() {
-        let path = test_path("tests/fixtures/C.jpg");
-        let data_dir = "target/data_dir";
-        if PathBuf::from(data_dir).exists() {
-            remove_dir_all(data_dir).unwrap();
-        }
-        let result = read_file(&path, Some(data_dir.to_owned()));
-        assert!(result.is_ok());
-        let json_report = result.unwrap();
-        println!("{}", json_report);
-        assert!(json_report.contains("C.jpg"));
-        assert!(PathBuf::from(data_dir).exists());
-        assert!(json_report.contains("thumbnail"));
-    }
-}
```

Comment:
Removes this because it wasn't being used in the new APIs, and we've removed the v1 and add_thumbnails features this was reliant upon for being included.